suspicion_test.go

```diff
@@ -0,0 +1,135 @@
+package memberlist
+
+import (
+	"testing"
+	"time"
+)
+
+func TestSuspicion(t *testing.T) {
+	const k = 3
+	const min = 500 * time.Millisecond
+	const max = 2 * time.Second
+
+	type pair struct {
+		from    string
+		newInfo bool
+	}
+	cases := []struct {
+		numConfirmations int
+		from             string
+		confirmations    []pair
+		expected         time.Duration
+	}{
+		{
+			0,
+			"me",
+			[]pair{},
+			max,
+		},
+		{
+			1,
+			"me",
+			[]pair{
+				pair{"me", false},
+				pair{"foo", true},
+			},
+			1250 * time.Millisecond,
+		},
+		{
+			1,
+			"me",
+			[]pair{
+				pair{"me", false},
+				pair{"foo", true},
+				pair{"foo", false},
+				pair{"foo", false},
+			},
+			1250 * time.Millisecond,
+		},
+		{
+			2,
+			"me",
+			[]pair{
+				pair{"me", false},
+				pair{"foo", true},
+				pair{"bar", true},
+			},
+			810 * time.Millisecond,
+		},
+		{
+			3,
+			"me",
+			[]pair{
+				pair{"me", false},
+				pair{"foo", true},
+				pair{"bar", true},
+				pair{"baz", true},
+			},
+			min,
+		},
+		{
+			3,
+			"me",
+			[]pair{
+				pair{"me", false},
+				pair{"foo", true},
+				pair{"bar", true},
+				pair{"baz", true},
+				pair{"zoo", false},
+			},
+			min,
+		},
+	}
+	for i, c := range cases {
```

Comment:
Might be easier to test if the struct split off the method of computing the timeout from the actual timer handling, then you can just test the logic instead of the timer interactions